keygen/CMakeLists.txt

```diff
@@ -5,15 +5,20 @@ project(enftun-keygen
   LANGUAGES C
 )
 
+option(KEYGEN_USE_TPM "Build keygen with TPM support" ON)
```

Comment:
It looks like the `USE_TPM` preprocessor define isn't actually used anywhere within the enftun code (ignoring keygen).

In the enftun code, ignoring the xtt code, there is no place that compiles/links against the TSS, either. The TPM stuff is dynamically loaded when the engine is.

So, from a build perspective, enftun can be built with `BUILD_XTT=OFF`, and still have TPM support (because the OpenSSL engine code gets built no matter what).

Behaviorally, the `enftun_tls_tpm_use_key` function will always get called in `enftun.c:97` if the file-based private key loading fails.

It seems like we wouldn't want to always use that function, only if someone has requested it. But, that should be a config option, not a compile-time option (since, again, it doesn't require any different libraries).

So, how about this:
- Remove the now-defunct `USE_TPM` definition in the top-level CMakeLists
- In `enftun_tls_load_credentials`, use the `SSL_CTX_use_PrivateKey_file` functions if `tpm_enable` is false in the config, and otherwise use `enftun_tls_tpm_use_key`?
  - That `tpm_enable` is a new (to this PR) config variable, set to true if the `tpm` block is present in the config (like the `xtt_enable` param)

I might be misunderstanding, since I'm still new to this project

Reply:
`USE_TPM` was probably defined because `xtt` needed it at one point, but xtt now properly handles its own defs, I think.

What about `enftun-keygen`?  Are you keeping the `KEYGEN_USE_TPM` option?  At the very least, that should be changed to follow the `BUILD`, not `USE`, naming convention.

I think it's still conceptually cleaner to have a `BUILD_TPM` option at the top-level (with `BUILD_XTT` implyi9ng `BUILD_TPM`).  The only feature currently gated by `BUILD_TPM` would be TPM support in `enftun-keygen`.    I agree that the "TPM support" in `enftun` should be gated by the `tpm_enable` config option, rather than the `BUILD_TPM` compile option.

Reply:
I just pushed a new commit, that cosolidates the old `KEYGEN_USE_TPM` into a single top-level `BUILD_TPM`

However, I now recant my assertion that the TPM load key should be gated by the config option. That should actually be "configured" in teh key file itself; that's why we perform that "is this a TPM key?" check before attempting to load it.

So, the commit I just pushed doesn't change that. And, because attempting to load a non-TPM key as a TPM key doesn't result in any confusing output nor does it require building against any other libraries, I also don't gate it with a compile-time `#define`

Reply:
Ok, that make sense.

> That should actually be "configured" in [the] key file itself; that's why we perform that "is this a TPM key?" check before attempting to load it.

The key file doesn't specify which TPM though, right?  The TPM config block is still needed to specify which TPM to use when a TPM is specified (or we need to modify the code to iterate over all available TPMs).

So using the presence of the TPM config block as a way for the user to disable TPM support (even if a TPM key file is provided) would be easy to implement.  We just don't see a reason to offer that level of  control, at this time.

Reply:
Yea, to your above comments.

My thought is basically Principle of Least Surprise. Within enftun, whether to use a TPM-resident key is set by what private key file is provided, *not* by a config option. So, if I configure the path to a TPM-resident private key file but forget to uncomment the `tpm` section of the config file, I might be confused why the handshake is failing